File: JobRecruitment/src/main/java/bachelor/thesis/job_recruitment/core/model/Background.java
package bachelor.thesis.job_recruitment.core.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.*;

import javax.persistence.*;

@EqualsAndHashCode(callSuper = false)
@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Builder
public class Background extends BaseEntity<Long>{
    private String techs;
    private String experience;
    private String jobType;
    private String devType;
    private Boolean remote;

}

File: JobRecruitment_Frontend/src/app/model/Background.ts
import {BaseEntity} from "./BaseEntity";
import {User} from "./User";

export class Background extends BaseEntity{
  techs: string;
  experience: string;
  jobType: string;
  devType: string;
  remote: boolean;


  constructor(id: number, techs: string, experience: string, jobType: string, devType: string, remote: boolean) {
    super(id);
    this.techs = techs;
    this.experience = experience;
    this.jobType = jobType;
    this.devType = devType;
    this.remote = remote;
  }
}
